app/views/user_checklists/show_progress.html.haml

```diff
@@ -0,0 +1,167 @@
+                  = complete_check_icon(html_options: { class: "fa-fw #{whiteout_class}", id: is_completed_icon_id })
+
+                = @user_checklist.name
+
+              = render partial: 'checklist_tree_as_list', locals: { user_checklist: @user_checklist }
+
+              %p.have-read-confirmation
+
+                = check_box_tag(checkbox_id, 'checked', @user_checklist.all_completed?,
+                    { class: "fa-check-square",
+                      onchange:"toggleCompletedCheckbox(#{@user.id}, #{@user_checklist.id})" })
+
+                = label_tag checkbox_id, { onchange:"toggleCompletedCheckbox(#{@user.id}, #{@user_checklist.id})" } do
+                  = t('.read_and_agree_start')
+                  %span.name= @user_checklist.name
+                  = t ('.read_and_agree_end')
+
+            .card-footer
+              .row
+                .col
+                  %p.contact-if-questions
+                    #contact-us-footer
+                      = t('.contact_if_questions')
+                      = mail_to(t('.email_display_name'), ENV['SHF_MEMBERSHIP_EMAIL'])
+
+                  - next_sibling = @user_checklist.next_sibling
+                  - if next_sibling
+                    .next
+                      = link_to  user_user_checklist_progress_path(@user, next_sibling), {id: next_link_id, class: hide_class } do
+                        = t('.next')
+                        %span.next-arrow-icon.show-inline#next-arrow-icon
+                          = next_arrow_icon(html_options: { class: 'fa-lg', id: next_arrow_id })
+
+
+:javascript
+
+  const setCompleteUrl = "/anvandare/lista/set-all-completed/"
+  const setUncompleteUrl = "/anvandare/lista/set-all-uncompleted/"
+
+  const checkboxId = "##{checkbox_id}";
+  const isCompleteIconId = "##{is_completed_icon_id}";
+  const innerProgressBarId = "##{inner_progress_bar_id}";
+  const percentTextClass = ".#{percent_text_class}";
+  const nextLinkId = "##{next_link_id}";
+  const contactUsFooterId = "##{contact_us_footer_id}";
+
+  const flexClass = "#{flex_class}";
+  const whiteoutClass = "#{whiteout_class}";
+  const showClass = "#{show_class}";
+  const hideClass = "#{hide_class}";
+
+  window.onload = function(){
+    updateBasedOnCompletedCheckbox();
+  }
+
+
+  function toggleCompletedCheckbox(userId, checklistId) {
+      let checkbox = $(checkboxId); //getCheckboxElement();
+      ajaxPostURL = (checkbox.prop("checked") ? setCompleteUrl : setUncompleteUrl)
+
+      $.post( ajaxPostURL + checklistId)
+
+        .done(function(responseData) {
+          if (Utility.checkHttpError(responseData) === false) {
+            // success
+            updateBasedOnCompletedCheckbox();
+            newPercentComplete = responseData.overall_percent_complete;
+            updateProgressBar(newPercentComplete);
+            if (newPercentComplete == 100) {
```

Comment:
If you're sure that `newPercentComplete` will always be in integer then it would be better to use `===`.